src/drivers/intel/hda/hda-dma.c

```diff
@@ -756,8 +759,8 @@ static int hda_dma_set_config(struct dma_chan_data *channel,
 	     config->src_width <= 2))
 		dgcs |= DGCS_SCS;
 
-	/* set DGCS.FIFORDY for output dma */
-	if (config->direction == DMA_DIR_MEM_TO_DEV || config->direction == DMA_DIR_LMEM_TO_HMEM)
+	/* set DGCS.FIFORDY for host DMA for both playback and capture */
+	if (config->direction == DMA_DIR_HMEM_TO_LMEM || config->direction == DMA_DIR_LMEM_TO_HMEM)
 		dgcs |= DGCS_FIFORDY;
 
 	dma_chan_reg_write(channel, DGCS, dgcs);
```

Comment:
Since we set/clear DGCS_FIFORDY explicitly in start/stop, do we need to preset it in _set_config() here?

Reply:
this seems to directly contradict the stated purpose of f96878a8f4e50e5b8023ddcf8810ea211b1899c3

Reply:
@lyakh I'm not sure what that specific commit achieved anyway. Before my change, we end up setting the FIFORDY bit for all DMAs unconditionally during START anyway.

Reply:
@ranj063 not sure GH made it clear enough what my comment was referring to. I meant the removal of cyclic support as per your commit comment "config->cyclic is not relevant for HDA DMAs." I'm not sure whether your comment is correct, but I just commented, that it contradicted the original commit, that introduced cyclic DMA support for HDA. Don't know whether that commit was right though.

Reply:
Maybe best to state in the comments the current feedback on setting ```DGCS.FIFORDY``` from KarL. This way everyone can understand.

Reply:
@lgirdwood I have updated the commit message and added comments in the next patch for FIFORDY bit updates.

Reply:
I think I was confused by the commit description phrase "`config->cyclic` is not relevant for HDA DMAs." It probably means "this one specific use of `config->cyclic` is irrelevant." If it really is trying to say, that the whole `config->cyclic` support in HDA DMA is irrelevant, then probably the whole commit https://github.com/thesofproject/sof/commit/f96878a8f4e50e5b8023ddcf8810ea211b1899c3 should be reverted.